### ISSUE TITLE
Bug: JSX namespace not being recognized in vite TSX context

### ISSUE THREAD

**Issue by aidenybai** [aidenybai/million#118]
### Describe the bug

```markdown
`JSX element implicitly has type 'any' because no interface 'JSX.IntrinsicElements' exists.` error with jsx and vite.
```

### To Reproduce

```markdown
1. Setup a vite project with this config: 

import { defineConfig } from 'vite';

export default defineConfig({
  esbuild: {
    jsxFactory: 'jsx',
    jsxFragment: 'Fragment',
    jsxInject: `import { jsx, jsxs, Fragment } from 'million/jsx-runtime'`,
  },
});

2. Create a `index.tsx` and `index.html`, and import the tsx file.
3. Write some jsx
4. View intellisense errors
```

### Expected behavior

```markdown
Should not have this error and vite should use the JSX namespace
```

### Screenshots

```markdown
N/A
```

### Operating System

MacOS Monterey

### Browser

Brave 1.29.77 Chromium: 93.0.4577.63 (Official Build) (arm64)

### Additional context

```markdown
N/A
```

**Comment by hyrious** [comment]
To let TypeScript find your jsx-runtime's types, there should be `{path}.d.ts` or `{path}/index.d.ts` where `path` = `million/jsx-runtime`.

So the solution is quite simple:
Move the `./dist/jsx-runtime.d.ts` to `./jsx-runtime.d.ts`.

- - -

What's more, users should know the correct way to write tsconfig.json:

```js
{
    "compilerOptions": {
        "jsx": "react-jsx",
        "jsxImportSource": "million",
        // below is required to find million from node_modules
        "moduleResolution": "Node"
    }
}
```

Also, the file containing jsx (tsx) should have at least one `import/export` to be recognized as es module:

```tsx
export type {} // <- this is required, if there's no other import/export
const hello = <div>Hello</div>
console.log(hello)
```

Just put the info here for completion.

**Comment by hyrious** [comment]
Also, there's no problem with your vite config, it is correct, and vite is able to build it correctly.

**Comment by aidenybai** [comment]
I'm wondering if theres a more elegant way to do this. Initially, I wanted the conditional export on [package.json L29](https://github.com/aidenybai/million/blob/6b794d355c0e7a60c47db3a0ca6349531d54e231/package.json#L29) to be "trapped" first (this one has types associated), and use the `jsx-runtime.js` as a fallback, but it seems like the `jsx-runtime.js` is being resolved first, thereby not being able to get the types. 

Is it a wise solution to just remove `jsx-runtime.js`, and hope the bundler is able to catch the conditional export?

**Comment by hyrious** [comment]
Let's clarify that here are two concepts:

- bundlers reads `exports` to resolve js file, which is exactly `dist/jsx-runtime.js`. They have no reason to choose `./jsx-runtime` because `exports` has the highest priority and once set, there can only be those paths to be exported.

- typescript reads `types` (for entry point) and tsconfig (if exist) to resolve types. Other paths are mapped to be related to the root of package folder.

That is to say, typescript does not know the `exports` field (at least now), since it already has `paths` in tsconfig.

(There's a deprecated field called typeVersion which can redirect type path, but I don't recommend that.)

In conclusion, here's a valid pkg stucture if your exports.jsx-runtime is set to ./dist/jsx-runtime.js:

```
./dist/jsx-runtime.js
./jsx-runtime.d.ts
```
Another one (also valid):
```
./dist/jsx-runtime.js
./jsx-runtime/index.d.ts
```
Yet another one:
```
./dist/jsx-runtime.js
./dist/jsx-runtime.d.ts
./jsx-runtime.d.ts
  ^ contents: export * from "./dist/jsx-runtime"
```

**Comment by aidenybai** [comment]
Thanks! Your explanation was really helpful. I chose to go with the:

```
./dist/jsx-runtime.js
./dist/jsx-runtime.d.ts
./jsx-runtime.d.ts
  ^ contents: export * from "./dist/jsx-runtime"
```

In the commit I referenced
